.github/workflows/mkdocs-deploy.yml

```diff
@@ -0,0 +1,47 @@
+name: build_and_deploy_mkdocs
+
+on:
+  push:
+    branches:
+      - main  # Triggers deployment on push to the main branch
+      - start-docs
+
+env:
+  UV_SYSTEM_PYTHON: true
+
+permissions:
+   contents: write
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Configure Git Credentials
+        run: |
+          git config user.name github-actions[bot]
+          git config user.email 41898282+github-actions[bot]@users.noreply.github.com
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+
+      - name: Cache mkdocs-material enviroment
+        uses: actions/cache@v3
+        with:
+          key: mkdocs-material-${{ env.cache_id }}
+          path: .cache
+          restore-keys: |
+            mkdocs-material-
+
+      - name: Install Dependencies
+        run: |
+          curl -LsSf https://astral.sh/uv/install.sh | sh
+          uv pip install ".[docs]"
+
+      - name: Build and Deploy
+        run: |
+          mkdocs gh-deploy --force --remote-branch gh-pages
```

Comment:
Could this not create the same problem we had on the nexus definitions, that the repository can grow arbitrarily large? Or will this not build up a history?

Reply:
This will overwrite the docs any time we make a merge to main, so the docs will not accumulate. We don't have docs for the individual branches